modules/fedimint-meta-client/src/lib.rs

```diff
@@ -0,0 +1,192 @@
+    pub async fn submit(&self, key: MetaKey, value: MetaValue) -> anyhow::Result<()> {
+        self.module_api
+            .submit(key, value, self.admin_auth()?)
+            .await?;
+
+        Ok(())
+    }
+
+    /// Get the current meta consensus value along with it's revision
+    ///
+    /// See [`Self::get_consensus_value_rev`] to use when checking for updates.
+    pub async fn get_consensus_value(
+        &self,
+        key: MetaKey,
+    ) -> anyhow::Result<Option<MetaConsensusValue>> {
+        Ok(self.module_api.get_consensus(key).await?)
+    }
+
+    /// Get the current meta consensus value revision
+    ///
+    /// Each time a meta consensus value changes, the revision increases,
+    /// so checking just the revision can save a lot of bandwidth in periodic
+    /// checks.
+    pub async fn get_consensus_value_rev(&self, key: MetaKey) -> anyhow::Result<Option<u64>> {
+        Ok(self.module_api.get_consensus_rev(key).await?)
+    }
+
+    /// Get current submissions to change the meta consensus value.
+    ///
+    /// Upon changing the consensus
+    pub async fn get_submissions(
+        &self,
+        key: MetaKey,
+    ) -> anyhow::Result<BTreeMap<PeerId, MetaValue>> {
+        Ok(self
+            .module_api
+            .get_submissions(key, self.admin_auth()?)
+            .await?)
+    }
+}
+
+/// Data needed by the state machine
+#[derive(Debug, Clone)]
+pub struct MetaClientContext {
+    pub meta_decoder: Decoder,
+}
+
+// TODO: Boiler-plate
+impl Context for MetaClientContext {}
+
+#[apply(async_trait_maybe_send!)]
+impl ClientModule for MetaClientModule {
+    type Init = MetaClientInit;
+    type Common = MetaModuleTypes;
+    type Backup = NoModuleBackup;
+    type ModuleStateMachineContext = MetaClientContext;
+    type States = MetaStateMachine;
+
+    fn context(&self) -> Self::ModuleStateMachineContext {
+        MetaClientContext {
+            meta_decoder: self.decoder(),
+        }
+    }
+
+    fn input_amount(
+        &self,
+        _input: &<Self::Common as ModuleCommon>::Input,
+    ) -> Option<TransactionItemAmount> {
+        unreachable!()
+    }
+
+    fn output_amount(
+        &self,
+        _output: &<Self::Common as ModuleCommon>::Output,
+    ) -> Option<TransactionItemAmount> {
+        unreachable!()
+    }
+
+    fn supports_being_primary(&self) -> bool {
+        false
+    }
+
+    async fn get_balance(&self, _dbtx: &mut DatabaseTransaction<'_>) -> Amount {
+        Amount::ZERO
+    }
+
+    #[cfg(feature = "cli")]
+    async fn handle_cli_command(
+        &self,
+        args: &[std::ffi::OsString],
+    ) -> anyhow::Result<serde_json::Value> {
+        cli::handle_cli_command(self, args).await
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct MetaClientInit;
+
+// TODO: Boilerplate-code
+#[apply(async_trait_maybe_send!)]
+impl ModuleInit for MetaClientInit {
+    type Common = MetaCommonInit;
+    const DATABASE_VERSION: DatabaseVersion = DatabaseVersion(2);
```

Comment:
I don't think this breaks anything, but should we start at 0?

Reply:
Yes. This is what I get for copying it from existing module. Good catch. Will follow-up.